dot/peerset/peerset_test.go

```diff
@@ -4,17 +4,78 @@
 package peerset
 
 import (
+	"sync"
 	"testing"
 	"time"
 
 	"github.com/libp2p/go-libp2p-core/peer"
 	"github.com/stretchr/testify/require"
 )
 
+type assertMessageParam struct {
+	expect interface{}
+	got    interface{}
+}
+
+type mockProcessMessage struct {
+	callCounter      int
+	executionCounter int
+	expects          map[int]assertMessageParam
+	mockMutex        sync.Mutex
+}
+
+func newMockProcessMessage() *mockProcessMessage {
+	return &mockProcessMessage{
+		expects: make(map[int]assertMessageParam),
+	}
+}
+
+func (mock *mockProcessMessage) ExpectByCall(expected interface{}) {
+	mock.mockMutex.Lock()
+	defer mock.mockMutex.Unlock()
+
+	mock.callCounter++
+	mock.expects[mock.callCounter] = assertMessageParam{
+		expect: expected,
+	}
+}
+
+func (mock *mockProcessMessage) ProcessMessage() func(Message) {
+	return func(m Message) {
+		mock.mockMutex.Lock()
+		defer mock.mockMutex.Unlock()
+
+		mock.executionCounter++
+		assert, has := mock.expects[mock.executionCounter]
+		if has {
+			assert.got = m
+			mock.expects[mock.executionCounter] = assert
+		}
+	}
+}
+
+func (mock *mockProcessMessage) Assert(t *testing.T) {
+	mock.mockMutex.Lock()
+	defer mock.mockMutex.Unlock()
+
+	// the number of calls to processMessage should be the same we expect
+	require.Equal(t, mock.callCounter, mock.executionCounter)
+
+	// follow the order we defined
+	for i := 1; i <= mock.callCounter; i++ {
+		assert, has := mock.expects[i]
+		require.True(t, has)
+		require.Equal(t, assert.expect, assert.got)
+	}
+}
+
 func TestPeerSetBanned(t *testing.T) {
```

Comment:
Since, test name is does not clearly say what behaviour we are testing we might want to add a comment